custom/benin/management/commands/migrate_users_and_their_cases_to_new_rc_level.py

```diff
@@ -0,0 +1,199 @@
+                            _update_cases(domain=domain, user=user, current_owner_id=village.location_id,
+                                          new_owner_id=new_user_rc_location.location_id,
+                                          dry_run=dry_run)
+                            _update_users_location(user=user, location=new_user_rc_location, dry_run=dry_run)
+                            log(f"User {user.username}:{user.user_id} updates completed.")
+                        else:
+                            log_error(f"User {user.username}:{user.user_id} rc {user_rc_number} location "
+                                      f"not found")
+                else:
+                    log_error(f"User {user.username}:{user.user_id} missing rc number")
+            log(f"Updates for village {village.name} processed.")
+
+
+def _find_locations(domain, location_type_code):
+    return SQLLocation.active_objects.filter(
+        domain=domain,
+        location_type__code=location_type_code
+    )
+
+
+def _find_rc_users_at_location(domain, location):
+    # return users with usertype as 'rc'
+    users = _find_users_at_location(domain, location)
+    prime_user_data_caches(users, domain)
+
+    return [
+        user
+        for user in users
+        if user.user_data.get('usertype') == 'rc'
+    ]
+
+
+def _find_users_at_location(domain, location):
+    return get_users_by_location_id(domain, location.location_id)
+
+
+def _find_child_location_with_name(parent_location, location_name):
+    # find location under parent location that has the name location_name
+    locations = parent_location.get_descendants().filter(
+        name=location_name
+    )
+    if not locations:
+        return None
+    if len(locations) == 1:
+        return locations[0]
+    if len(locations) > 1:
+        raise MultipleMatchingLocationsFound
+
+
+def _update_cases(domain, user, current_owner_id, new_owner_id, dry_run):
+    case_types = ['menage', 'membre', 'seance_educative', 'fiche_pointage']
+    for case_type in case_types:
+        case_ids = _find_case_ids(case_type=case_type, owner_id=current_owner_id, opened_by_user_id=user.user_id)
+
+        log(f"fUpdating {len(case_ids)} {case_type} cases for user {user.username}")
+
+        for case_ids in with_progress_bar(
+            chunked(case_ids, 100),
+            length=math.ceil(len(case_ids) / 100),
+            oneline=False
+        ):
+            _update_case_owners(domain, case_ids, new_owner_id, dry_run)
+
+
+def _update_case_owners(domain, case_ids, owner_id, dry_run):
+    case_blocks = []
+    for case_id in case_ids:
+        case_blocks.append(
+            CaseBlock(
+                create=False,
+                case_id=case_id,
+                owner_id=owner_id
+            )
+        )
+    if not dry_run:
+        submit_case_blocks(
+            case_blocks=case_blocks,
+            domain=domain,
+            device_id=__name__ + ".migrate_users_and_their_cases_to_new_rc_level"
```

Comment:
Wanted to confirm if this is the standard way to name the device id for updates through scripts ?
The device id here would be `custom/benin/management/commands/migrate_users_and_their_cases_to_new_rc_level.migrate_users_and_their_cases_to_new_rc_level`

Reply:
ya thats too long. You could see other places how its done.

Reply:
Quick checking at other places and this does look like the right way for adhoc updates, I am gonna just skip the '.migrate_users_and_their_cases_to_new_rc_level' as it is repetitive

Reply:
okay.
I thought device ID would be `custom.benin.management.commands.migrate_users_and_their_cases_to_new_rc_level`. Surprised to see the slash.

When we use the same approach [for imports](https://github.com/dimagi/commcare-hq/blob/b7afe8e660648f088085f64a0e51c252f281c668/corehq/apps/case_importer/do_import.py#L489) the device id is set to `corehq.apps.case_importer.do_import.do_import `.

Reply:
> I thought device ID would be custom.benin.management.commands.migrate_users_and_their_cases_to_new_rc_level. Surprised to see the slash.

Sorry, my bad. There would be no slashes. It will be `.` instead of slashes.